optimum/exporters/executorch/tasks/multimodal_text_to_text.py

```diff
@@ -0,0 +1,179 @@
+            f"{POTENTIAL_DECODER_NAMES}. This is required for multimodal text-to-text models."
+        )
+
+    # Find encoder components
+    audio_encoder_name = None
+    for name in POTENTIAL_AUDIO_ENCODER_NAMES:
+        if hasattr(model, name):
+            audio_encoder_name = name
+            break
+
+    vision_encoder_name = None
+    for name in POTENTIAL_VISION_ENCODER_NAMES:
+        if hasattr(model, name):
+            vision_encoder_name = name
+            break
+
+    if (audio_encoder_name is None) == (vision_encoder_name is None):
+        raise ValueError(
+            "The model does not have one of the expected encoder attributes: "
+            f"{POTENTIAL_AUDIO_ENCODER_NAMES + POTENTIAL_VISION_ENCODER_NAMES}. "
+            "This is required for multimodal text-to-text models."
+            "Currently only a maximum of 1 modality is supported, so there can only be one of these"
+            "encoders in the model."
+        )
+
+    return decoder_name, audio_encoder_name, vision_encoder_name
+
+
+# NOTE: It's important to map the registered task name to the pipeline name in https://github.com/huggingface/transformers/blob/main/utils/update_metadata.py.
+# This will streamline using inferred task names and make exporting models to Hugging Face pipelines easier.
+@register_task("image-text-to-text")
+@register_task("audio-text-to-text")
+@register_task("multimodal-text-to-text")  # Fake task name, since audio-text-to-text is not available.
+def load_multimodal_text_to_text_model(model_name_or_path: str, **kwargs):
+    """
+    Loads a causal language model for multimodal generation (e.g. image-to-text) generation and registers it under the appropriate task
+    (e.g. 'image-text-to-text') using Hugging Face's AutoModelForCausalLM.
+
+    Args:
+        model_name_or_path (str):
+            Model ID on huggingface.co or path on disk to the model repository to export. For example:
+            `model_name_or_path="google/gemma-3-4b-it"` or `model_name_or_path="/path/to/model_folder`
+        **kwargs:
+            Additional configuration options for the model:
+                - dtype (str, optional):
+                    Data type for model weights (default: "float32").
+                    Options include "float16" and "bfloat16".
+                - attn_implementation (str, optional):
+                    Attention mechanism implementation (default: "sdpa").
+                - cache_implementation (str, optional):
+                    Cache management strategy (default: "static").
+                - max_length (int, optional):
+                    Maximum sequence length for generation (default: 2048).
+
+    Returns:
+        MultiModalTextToTextExportableModule:
+            An instance of `MultiModalTextToTextExportableModule` for exporting and lowering to ExecuTorch.
+    """
+    device = "cpu"
+    batch_size = 1
+    dtype = kwargs.get("dtype", "float32")
```

Comment:
is dtype really somethng we allow to override? use_custom_sdpa doesnt work with fp16